src/main/java/org/squbs/sample/SampleHttpService.java

```diff
@@ -0,0 +1,106 @@
+package org.squbs.sample;
+
+import akka.NotUsed;
+import akka.http.javadsl.marshallers.jackson.Jackson;
+import akka.http.javadsl.model.ContentTypes;
+import akka.http.javadsl.model.StatusCodes;
+import akka.http.javadsl.model.headers.RawHeader;
+import akka.http.javadsl.server.PathMatchers;
+import akka.http.javadsl.server.Route;
+import akka.japi.pf.PFBuilder;
+import akka.stream.javadsl.Source;
+import akka.util.ByteString;
+import akka.util.Timeout;
+import org.squbs.actorregistry.japi.ActorLookup;
+import org.squbs.unicomplex.AbstractRouteDefinition;
+import scala.concurrent.duration.Duration;
+import scala.concurrent.duration.FiniteDuration;
+
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+
+import static akka.http.javadsl.model.HttpEntities.createChunked;
+import static akka.http.javadsl.server.PathMatchers.segment;
+import static scala.compat.java8.JFunction.func;
+
+/**
+ * The route definition.
+ */
+public class SampleHttpService extends AbstractRouteDefinition {
+
+    final ActorLookup<?> lookup = ActorLookup.create(context());
+    final Timeout timeout = new Timeout(Duration.create(5, TimeUnit.SECONDS));
+
+    private Route anonymous = route(
+        path("hello", () -> pathEnd(() -> {
+                CompletionStage<String> respCS =
+                    lookup.ask(new PingRequest("anonymous"), timeout)
+                        .thenApply(PingResponse.class::cast)
+                        .thenApply(o -> o.message);
+
+                if (respCS.toCompletableFuture().isCompletedExceptionally()) {
```

Comment:
I'll revisit this route as well to use `onComplete()`.  I'd like to confirm the usage is correct though in the other routes here.